packages/core/src/components/select/select.tsx

```diff
@@ -48,6 +50,77 @@ export class AtomSelect {
   @Event() atomDismiss!: EventEmitter<void>
   @Event() atomFocus!: EventEmitter<void>
 
+  @Method()
+  setTag() {
+    /**
+     * This method was necessary because the `ion-selection-option` loop does not allow customizations or custom components.
+     * So, to be able to add custom elements such as a tag or a badge inside an option of the `select` field, when the select
+     * is opened, the `onBlur` event triggers this method that performs a search for all `ion-item` elements (which is the
+     * final element rendered to list options) and filters the ones that need to be changed.
+     */
+
+    const ionItemElements = document.querySelectorAll('ion-item')
+
+    ionItemElements?.forEach((itemElement) => {
```

Comment:
I don't know why TS does not recognize the `itemElement`, but you can add it manually: 

`    ionItemElements?.forEach((itemElement: HTMLIonItemElement) => {
`

Reply:
@tassioFront document.querySelectorAll we can pass a type parameter to specify element
```suggestion
const ionItemElements = document.querySelectorAll<HTMLIonItemElement>('ion-item')
```

Reply:
perfect, @igorwessel !